dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -210,6 +213,20 @@ public Response toolsIndexGet(SecurityContext securityContext) {
     @Override
     public Response toolsIndexSearch(String query, MultivaluedMap<String, String> queryParameters, SecurityContext securityContext) {
         if (!config.getEsConfiguration().getHostname().isEmpty()) {
+            // Performing a search on the UI sends multiple POST requests. When the search term ("include" key in request payload) is large,
+            // one of these POST requests will fail, but the others will continue to pass.
+            JSONObject json = new JSONObject(query);
+            try {
+                String include = json.getJSONObject("aggs").getJSONObject("autocomplete").getJSONObject("terms").getString("include");
```

Comment:
Instead of `getString("include")`, I think you could make it `optString("include")`, which will return an empty string if the property doesn't exist. Then you wouldn't need the try/catch.